src/Confix.Tool/src/Confix.Tool/Utilities/Json/VariableIntellisenseRewriter.cs

```diff
@@ -0,0 +1,41 @@
+using System.Text.Json.Nodes;
+using Json.More;
+using Json.Schema;
+
+namespace Confix.Tool.Schema;
+
+public readonly record struct VariableIntellisenseContext(
+    JsonObject VariableReference
+);
+
+/// <summary>
+/// This rewriter is used to replace all primitives with a type allowing either the primitive
+/// or a variable. This is used to allow the user to specify a variable with the support
+/// of IntelliSense.
+/// </summary>
+public sealed class VariableIntellisenseRewriter : JsonDocumentRewriter<VariableIntellisenseContext>
+{
+    protected override JsonNode Rewrite(JsonObject obj, VariableIntellisenseContext context)
+    {
+        string? typeName = obj
+            .Where(x => x.Key == TypeKeyword.Name)
+            .Select(x => (string)x.Value!)
```

Comment:
can we not just use interpolation here instead of the cast :D 
```suggestion
            .Select(x => $"{x}")
```